server/internal/cache/blob/cache.go

```diff
@@ -314,11 +326,7 @@ func (c *DiskCache) Unlink(name string) error {
 	if err != nil {
 		return err
 	}
-	err = os.Remove(manifest)
-	if errors.Is(err, fs.ErrNotExist) {
-		return nil
-	}
-	return err
+	return os.Remove(manifest)
```

Comment:
Revert back to nil and do a check in the http handler instead. Internally we don't care if the result is a nop, we only care that the manifest by name is non existent when Unlink returns.

Alternatively, it could return (ok bool, _ error) so that calls looking for signs that it did exist during the call will have that feedback.